pennylane/devices/default_mixed.py

```diff
@@ -335,6 +335,50 @@ def _apply_channel(self, kraus, wires):
 
         self._state = qnp.einsum(einsum_indices, kraus, self._state, kraus_dagger)
 
+    def _apply_channel_tensordot(self, kraus, wires):
+        r"""Apply a quantum channel specified by a list of Kraus operators to subsystems of the
+        quantum state. For a unitary gate, there is a single Kraus operator.
+
+        Args:
+            kraus (list[array]): Kraus operators
+            wires (Wires): target wires
+        """
+        channel_wires = self.map_wires(wires)
+        num_ch_wires = len(channel_wires)
+
+        # Shape kraus operators and cast them to complex data type
+        kraus_shape = [2] * (num_ch_wires * 2)
+        kraus = [qnp.cast(qnp.reshape(k, kraus_shape), dtype=self.C_DTYPE) for k in kraus]
+
+        # row indices of the quantum state affected by this operation
+        row_wires_list = channel_wires.tolist()
+        # column indices are shifted by the number of wires
+        col_wires_list = [w + self.num_wires for w in row_wires_list]
+
+        channel_col_ids = list(range(num_ch_wires, 2 * num_ch_wires))
+        axes_left = [channel_col_ids, row_wires_list]
+        # Use column indices instead or rows to incorporate transposition of K^\dagger
+        axes_right = [col_wires_list, channel_col_ids]
+        # Apply the Kraus operators, and sum over all Kraus operators afterwards
+        _state = qnp.sum(
+            qnp.stack(
+                [
+                    qnp.tensordot(qnp.tensordot(k, self._state, axes_left), qnp.conj(k), axes_right)
+                    for k in kraus
+                ]
+            ),
+            axis=0,
+        )
+
+        # Permute the affected axes to their destination places.
+        # The row indices of the kraus operators are moved from the beginning to the original
+        # target row locations, the column indices from the end to the target column locations
+        source_left = list(range(num_ch_wires))
+        dest_left = row_wires_list
+        source_right = list(range(-num_ch_wires, 0))
+        dest_right = col_wires_list
+        self._state = qnp.moveaxis(_state, source_left + source_right, dest_left + dest_right)
```

Comment:
Not sure but is it necessary if you don't map the wires at the beginning?

Reply:
Nice question! :) I _think_ it's two separate "transpositions". 
1. the mapping in the beginning leads to the matrix multiplication contracting over the correct indices.
2. The `moveaxis` here pushes the new axis into the right position, which is not done by `tensordot` automatically. An example:
Consider a 3-tensor `T` of shape `(a, b, c)` and say we want to multiply the second axis with a `(b, b)` matrix `M`. Then `S=tensordot(M, T, axes=[1, 1])` will do the multiplication, but the result has shape `(b, a, c)`, because the `0`th axis of `M` remains at the left-most position, and the the axes indicated in the `axes` kwarg have been contracted. In order to restore the original shape, we have to do `moveaxis(S, 0, 1)`, which `einsum` can just do based on the position of the indices written on the right-hand side of the `"->"`.

Does this make sense? Or do you nonetheless see a way of skipping transposing wires? :)

Reply:
Great explanation, all good on my side!